pkg/apis/v1beta1/nodepool.go

```diff
@@ -95,6 +95,22 @@ type Disruption struct {
 	// +kubebuilder:validation:MaxItems=50
 	// +optional
 	Budgets []Budget `json:"budgets,omitempty" hash:"ignore"`
+	// TerminationGracePeriod is the duration the controller will wait before forcefully terminating a node, measured from when deletion is first initiated.
+	// Once the GracePeriod has expired, all pods on the node will be shutdown using the official non-graceful shutdown taint.
+	//
+	// Warning: this feature takes precedence over a Pod's terminationGracePeriodSeconds value, and bypasses any blocked PDBs or the karpenter.sh/do-not-disrupt annotation.
+	//
+	// This field is intended to be used by cluster administrators to enforce that nodes can be cycled within a given time period.
+	// When set, expired and drifted nodes will begin draining even if there are pods blocking eviction. Draining will respect PDBs and do-not-disrupt until the TGP is reached.
```

Comment:
Expired will need to be dropped with once rebased since expiration is now forceful. Same comment for the v1 version.